### ISSUE TITLE
System Statistics tab not working 

### ISSUE THREAD

**Issue by satvvik** [gduvalsc/kairos#1]
Hi,

The System Statistics tab is not working on 12.2.0.1. I deeply appreciate your work and it helps me a lot with my analysis.

Regards,
Satvvik

**Comment by satvvik** [comment]
<img width="927" alt="Error" src="https://user-images.githubusercontent.com/56820923/67190740-22148380-f40e-11e9-9c21-c560db1bcb78.png">

**Comment by gduvalsc** [comment]
Hi satvvik,

Thank you for using Kairos!

Could you attach the zip file you have uploaded into Kairos? It will be easy for me to understand the problem? I will provide you a solution very quickly!

Regards.

Gerard

**Comment by gduvalsc** [comment]
Satvvik,

It seems that you have used the kairos script (provided in the collects directory), but did you require an AWR report in HTML format or not ?

Generally when I use this script I always require a report in HTML format.

From my side, I already used this collect script with Oracle 12.2.0.1 and don't have any problem to display the system statistics.

This is why I need your "zip" file to investigate this problem.

Regards.

Gerard

**Comment by satvvik** [comment]
Thanks Gerard for quick response. You are right, I used the kairos script to gather data but I too, used the html format for Oracle DB v12.2.0.1.

I am awaiting internal approvals to send you the zip file. I'll post the same as soon as I get the necessary approvals. Regret the delay.. Please bear with me.

Regards,
Satvvik Visswanathan

**Comment by gduvalsc** [comment]
Satvvik,

I’m waiting your test case when you’ll get the approvals.

For your information, I need to deliver a new Kairos version because Oracle has changed some tables between 11.2 and 12.2.

The problem is the following: there are 2 tables “Key Instance Activity stats” and “Instance Activity stats” in both version. In 11.2 statistics are either in one table or in the other table (but non in both). In 12.2 statistics are in both tables. I discovered this problem during a performance analysis last week (sessions logical reads and physical reads are wrong when Oracle version is 12.2)

The solution for this problem will be delivered with Kairos 6.2

Regards.

Gerard

De : satvvik <notifications@github.com>
Répondre à : gduvalsc/kairos <reply@reply.github.com>
Date : lundi 21 octobre 2019 à 14:17
À : gduvalsc/kairos <kairos@noreply.github.com>
Cc : Gérard Duval <gduvalsc@gmail.com>, Comment <comment@noreply.github.com>
Objet : Re: [gduvalsc/kairos] System Statistics tab not working (#1)


Thanks Gerard for quick response. You are right, I used the kairos script to gather data but I too, used the html format for Oracle DB v12.2.0.1.

I am awaiting internal approvals to send you the zip file. I'll post the same as soon as I get the necessary approvals. Regret the delay.. Please bear with me.

Regards,
Satvvik Visswanathan

**Comment by satvvik** [comment]
Thanks Gerard for the info. I have got the necessary approvals...

Uploading the zip here. I get the same error while going through segments by row lock waits section also. Request you to check and provide a fix.

Thanks !!
Satvvik
[kairos_BAAN_AM3P1_2019-09-16,1.zip](https://github.com/gduvalsc/kairos/files/3753899/kairos_BAAN_AM3P1_2019-09-16.1.zip)

**Comment by satvvik** [comment]
Gerard, 

We also are planning to upgrade a couple of databases to Oracle v18. Are you planning to support Oracle v18 on KAIROS. I can see in the collect script that you have only coded till 12.2.

Regards,
Satvvik

**Comment by gduvalsc** [comment]
Hello Satvvik,

Some explanations for you in order to be able to identify where the problem is…

In your case when you have such an error, the best is to investigate the kairos.log.

[Une image contenant capture d’écran  Description générée automatiquement]

So here, the problem comes from the query DBORASYS$$4

In this case, it’s necessary to check the text of the query:

[Une image contenant capture d’écran  Description générée automatiquement]

[Une image contenant capture d’écran  Description générée automatiquement]

[Une image contenant capture d’écran  Description générée automatiquement]

[Une image contenant capture d’écran  Description générée automatiquement]

You need to check the request. In the current case the request is:

SELECT timestamp,
       label      AS label,
       Sum(value) AS value
FROM   (SELECT timestamp,
               statistic                      AS label,
               value * 1.0 / 100 / avgelapsed AS value
        FROM   (SELECT t.timestamp,
                       statistic,
                       value,
                       avgelapsed
                FROM   dboraoss AS t,
                       dboramisc AS m
                WHERE  t.timestamp = m.timestamp
                       AND t.kairos_nodeid = m.kairos_nodeid) AS foo
        WHERE  statistic IN ( 'IOWAIT_TIME' )) AS foo
GROUP  BY timestamp,
          label
ORDER  BY timestamp

Here the request expects to find something related to “IOWAIT_TIME” in the table ORAOSS.

However in your case, when you are looking  at an HTML report there is no “IOWAIT_TIME” in the section related to Operating system statistics.

An example coming from your zip file:

[Une image contenant capture d’écran  Description générée automatiquement]
No IOWAIT_TIME entry!

An example coming from my report:

[Une image contenant capture d’écran  Description générée automatiquement]

I think the problem comes from the platform

[Une image contenant capture d’écran  Description générée automatiquement]

The best is to update the chart and to remove the part related to “IOWAIT_TIME”

Kairos is OPEN and you can do that in the section “Manage objects”:

[Une image contenant capture d’écran  Description générée automatiquement]

[Une image contenant capture d’écran  Description générée automatiquement]

You need to find the part of the chart related to your problem (the part including IOWAIT_TIME)

[Une image contenant capture d’écran  Description générée automatiquement]


[Une image contenant capture d’écran  Description générée automatiquement]


Now you have a new chart “DBORASYS” in your repository and this new chart overrides the old “DBORASYS” chart.

Your result is now the following:

[Une image contenant capture d’écran, texte  Description générée automatiquement]

Please confirm you will receive this message.

Regards.

Gerard

De : satvvik <notifications@github.com>
Répondre à : gduvalsc/kairos <reply@reply.github.com>
Date : mardi 22 octobre 2019 à 07:11
À : gduvalsc/kairos <kairos@noreply.github.com>
Cc : Gérard Duval <gduvalsc@gmail.com>, Comment <comment@noreply.github.com>
Objet : Re: [gduvalsc/kairos] System Statistics tab not working (#1)


Thanks Gerard for the info. I have got the necessary approvals...

Uploading the zip here. I get the same error while going through segments by row lock waits section also. Request you to check and provide a fix.

Thanks !!
Satvvik
kairos_BAAN_AM3P1_2019-09-16,1.zip<https://github.com/gduvalsc/kairos/files/3753899/kairos_BAAN_AM3P1_2019-09-16.1.zip>

**Comment by satvvik** [comment]
Hi Gerard, Thanks for your response, however, I am unable to see the images you posted.. Would it be possible for you to send the response again with the images ? 
It would really help

Thanks & Regards,
Satvvik

**Comment by gduvalsc** [comment]
Satvvik

The answer (in PDF format) in attachment

Regards.

Gerard

De : satvvik <notifications@github.com>
Répondre à : gduvalsc/kairos <reply@reply.github.com>
Date : mardi 22 octobre 2019 à 11:26
À : gduvalsc/kairos <kairos@noreply.github.com>
Cc : Gérard Duval <gduvalsc@gmail.com>, Comment <comment@noreply.github.com>
Objet : Re: [gduvalsc/kairos] System Statistics tab not working (#1)


Hi Gerard, Thanks for your response, however, I am unable to see the images you posted.. Would it be possible for you to send the response again with the images ?
It would really help

Thanks & Regards,
Satvvik

**Comment by gduvalsc** [comment]
Yes, of course, I’m planning to support Oracle V18 and later (especially Oracle 19c which is the long term support version) on Kairos.

The best is to try with the current script and you tell me if something wrong is occurring.

Regards.

Gerard

De : satvvik <notifications@github.com>
Répondre à : gduvalsc/kairos <reply@reply.github.com>
Date : mardi 22 octobre 2019 à 07:20
À : gduvalsc/kairos <kairos@noreply.github.com>
Cc : Gérard Duval <gduvalsc@gmail.com>, Comment <comment@noreply.github.com>
Objet : Re: [gduvalsc/kairos] System Statistics tab not working (#1)


Gerard,

We also are planning to upgrade a couple of databases to Oracle v18. Are you planning to support Oracle v18 on KAIROS. I can see in the collect script that you have only coded till 12.2.

Regards,
Satvvik
